include/swift/Option/FrontendOptions.td

```diff
@@ -486,4 +486,7 @@ def disable_verify_exclusivity : Flag<["-"], "disable-verify-exclusivity">,
 def enable_key_path_resilience : Flag<["-"], "enable-key-path-resilience">,
   HelpText<"Enable key path resilience.">;
 
+def use_dwarf_debug_flags : Flag<["-"], "use-dwarf-debug-flags">,
+  HelpText<"Write the compilation flags into the DWARF debug info.">;
```

Comment:
Perhaps: `-emit-debug-options: Emit the compiler invocation in the debug info.`

Reply:
That alternative seems fine; my only concern would be whether it would be easily confused with `-serialize-debugging-options`. Or do you think the symmetry between "serialize" and "emit" would intentionally work well and make it clear which one goes where?

Reply:
I don't love either name; it's not the "debug flags" or "debug options" that are being written. (It also doesn't seem DWARF-specific.) I'll toss another contribution in, `-enable-invocation-in-debug-info`, but that's getting fairly wordy.

Reply:
I think it would be nice if the debug-info part came first for grouping and sorting.
`-debug-info-(emit|store)-invocation`, `-debug-info-invocation=yes`?

Reply:
If we're making it a driver flag based on the conversation below, should we prefix it with `-g` then (but not add it to the `g_Group`)? How do any of these sound:

* `-gcompiler-flags`
* `-gcompiler-options`
* `-gcompiler-invocation`
* `-grecord-compiler-flags`
* `-grecord-compiler-options`
* `-grecord-compiler-invocation`

Reply:
\*sigh\* I'm generally against prefixed options but I think I'm losing that battle. Adrian?

Reply:
Do you have a good reason for why you don't like prefixed options? My main motivation was that they would sort naturally in `swift --help` and `swift (1)` (if we ever got around to actually writing a man page 🙄).

Reply:
Sorry, I'm fine with `-debug-info`; it's grouping a bunch of things under single letters that I don't like. To compare with GCC/Clang flags:

- -f* - something to do with the language? Why -f?
- -m* - something to do with code generation? Why -m?
- -d* - something to do with dependencies? Or "defines"?
- -i* - "includes", okay…but it led us to the horrible `-iframework`, meaning "system framework include path"

-g is at least consistently used for debugging things, but there's still no obvious link between "-g" and "debug info"; you just have to memorize it. -O, -W, and -X are pretty much the only flag families that stayed consistent, and -W is still weird around the edges.

Reply:
Also we do have a man page! It just tells you to use `-help`. :-)

Reply:
Alright, given the feedback so far, if we exclude prefixed flags, I think `-debug-info-store-invocation` is the one that best balances conciseness while still getting across what the flag actually does. `-enable-invocation-in-debug-info` came close, but I agree that it's a bit verbose.

If that sounds good, I'll go ahead and update the PR to use that.

Reply:
@jrose-apple the prefixes actually have some level of consistency

`-f`: feature flags
`-m`: machine flags
`-d`: dump flags (used for the C Preprocessor)
`-I`: includes

Reply:
Except that "feature flag" vs. "machine flag" doesn't make any sense to a normal user, and "-i", while consistent, has led to badness elsewhere.